### PR TITLE
Multi-Variable Propagation and Simplification

### PR DESCRIPTION
**Pull request by DavidBakerEffendi**
@PhillipVH @Monkleys @Zhunaidm @steynvl
Travis: https://travis-ci.org/DavidBakerEffendi/green/builds/415536124

### CODE REVIEW

File: src/za/ac/sun/cs/green/service/simplify/ConstantPropagation.java [hunk 1]
```diff
@@ -0,0 +1,952 @@
...
+import za.ac.sun.cs.green.expr.Variable;
+import za.ac.sun.cs.green.expr.Visitor;
+import za.ac.sun.cs.green.expr.VisitorException;
+import za.ac.sun.cs.green.service.BasicService;
+import za.ac.sun.cs.green.util.Reporter;
+
+public class ConstantPropagation extends BasicService {
+
+	/**
+	 * Number of times the slicer has been invoked.
```
Comment (PhillipVH) [inline src/za/ac/sun/cs/green/service/simplify/ConstantPropagation.java]: Probably want to update the name of the tool here.

File: src/za/ac/sun/cs/green/service/simplify/ConstantPropagation.java [hunk 2]
```diff
@@ -0,0 +1,952 @@
...
+		return result;
+	}
+
+	/**
+	 * Propagates variables from given equality expressions and simplifies
+	 * equation to attempt to reveal variable's values according to each
+	 * equation or true/false if {@link za.ac.sun.cs.green.expr.Expression
+	 * Expression} is boolean.
+	 * 
+	 * @param expression
```
Comment (PhillipVH) [inline src/za/ac/sun/cs/green/service/simplify/ConstantPropagation.java]: Javadoc! Praise the heavens :zap: :+1:

File: src/za/ac/sun/cs/green/service/simplify/ConstantPropagation.java (hunk 1 above)
Comment (DavidBakerEffendi) [reply src/za/ac/sun/cs/green/service/simplify/ConstantPropagation.java]: Will do shortly :shipit:

Comment (DavidBakerEffendi) [reply src/za/ac/sun/cs/green/service/simplify/ConstantPropagation.java]: Would SATSimplificationService be sufficient?

Comment (PhillipVH) [reply src/za/ac/sun/cs/green/service/simplify/ConstantPropagation.java]: Eh, GitHub review sucks. The window size is fixed, I was referring to the slicer in the comment.

**Pull request closed without merging**